WordPress/gutenberg
# Move `duplicateTemplatePart` action to the `@wordpress/fields` package

## What?



This PR migrates the `duplicateTemplatePart` action to the `@wordpress/fields` package. The primary issue with this action is its dependency on the `CreateTemplatePartModalContents` component, which resides in the `@wordpress/editor` package. Due to this, it can't be directly used within the `@wordpress/fields` package.

For the purpose of the `@wordpress/fields` package work, I have temporarily moved the component to this package. However, I'm not certain if this is the best long-term solution. Ideally, this component (and potentially others) should reside in a dedicated package, such as `@wordpress/template`, which would be more appropriate for components related to template and template-parts. A similar package for the patterns already exists: https://github.com/WordPress/gutenberg/blob/d6fcf4aea4aff17cf0910862bf10090e20ba2ab0/packages/patterns/src

Please review and provide feedback on whether this approach is suitable or if there are alternative recommendations for handling these dependencies. Thanks! 🙏 cc @youknowriad @oandregal 



## Testing Instructions





Ensure that `Custom Dataviews` is enabled.

1. Visit the Site Editor.
2. Visit the patterns view.
3. Click on header
4. Duplicate a header template part.
5. Ensure that the duplicated template part is created.

### Testing Instructions for Keyboard


## Screenshots or screencast
